packages/react/navigation-menu/src/NavigationMenu.tsx

```diff
@@ -504,6 +504,13 @@ const NavigationMenuLink = React.forwardRef<NavigationMenuLinkElement, Navigatio
           aria-current={active ? 'page' : undefined}
           {...linkProps}
           ref={forwardedRef}
+          onClick={composeEventHandlers(props.onClick, (event) => {
+            const rootContentDismissEvent = new Event(ROOT_CONTENT_DISMISS, {
```

Comment:
Given the changes in #1292 I wonder if it'll be worth at some point creating a reusable dispatch event function which uses `flushSync` or if it's better to keep the usage local, and case by case when we see issues. This is my biggest struggle, not knowing if it's needed or not 😀